prov/shm/src/smr_ep.c

```diff
@@ -1696,6 +1696,12 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			if (ret)
 				return ret;
 		} else {
+			srx = calloc(1, sizeof(*srx));
+			srx->peer_ops = &smr_srx_peer_ops;
+			srx->owner_ops = smr_get_peer_srx(ep)->owner_ops;
+			srx->ep_fid.fid.context =
+				smr_get_peer_srx(ep)->ep_fid.fid.context;
+			ep->srx = &srx->ep_fid;
```

Comment:
I know this is my bug and this fix came from my recent srx PR that I closed and I think it's a solution but I'm wondering if it would be better to move this to where we bind the srx to the ep [here](https://github.com/ofiwg/libfabric/blob/main/prov/shm/src/smr_ep.c#L1036)? I think it's weird that we set the ep->srx and then override it. Would probably be better to just set it accordingly in the first place? What do you think?

Reply:
That sounds better to me, will update, thank you!

Reply:
Thanks!

Reply:
Updated, I need to move smr_ep_bind after the definition of smr_srx_peer_ops.